### ISSUE TITLE
Ram debug mode

### ISSUE THREAD

**Issue by bfrog** [imxrt-rs/imxrt-rt#4]
It was clear from my experience the other day that the default linker mode should the same as the mcuxpresso sdk, putting everything in ram for ease of debugging. I've started a branch that effectively takes cortex-m-rt and ports it to imxrt.

Some things I still noticed, that are perhaps causing me some headaches and need to look into.

After a short period of time the program seems to jump to a random address. I'm guessing this is actually the mpu, watchdogs, or the systick interrupt being in the wrong place in play as it looks like from a combination of zephyr and mcuxpresso code, the watchdogs, mpu, icache and dcache all should be setup. I need to dig deeper to find out.

**Comment by bfrog** [comment]
I think I've narrowed this down to the VTOR register not being set appropriately in my branch.

**Comment by bfrog** [comment]
Doesn't seem to be the VTOR issue

**Comment by bfrog** [comment]
Disabling the watchdogs fixed jumping to a random address, I'm able to run code just fine now.

**Comment by bfrog** [comment]
This is in progress as the rust-runtime branch, I'm probably tempting fate by trying to move towards a rust powered runtime over the C runtime, but in looking at what we need to do, it seems like its entirely possible. I can always rewrite some critical sections in C or assembly based on mcuxpresso's startup (BSD licensed!) files.

**Comment by Florob** [comment]
It looks to me like the current implementation actually supports this when `RuntimeBuilder::flash_opts` is `None`. There just is no way to actually construct it like that. Is there a reason for this? Some missing implementation detail or know issue?

**Comment by mciantyre** [comment]
That's right; there's consideration for it in the implementation. Setting the flash options to `None` should execute all the code paths.

I had ambition to include a `from_ram` constructor in the initial release. But then I never tested it. There's no `inspect_elf` test for a RAM-only image, and I never loaded a RAM-only image onto any of my boards. I can't say for sure if it works.
